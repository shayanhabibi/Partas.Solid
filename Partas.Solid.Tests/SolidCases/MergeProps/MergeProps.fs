module Partas.Solid.Tests.SolidCases.MergeProps

open Partas.Solid
open Fable.Core

type [<Erase>] SimpleMerge() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.SimpleMerge =
        props.class' <- "DefaultClass"
        div()
        
type [<Erase>] MultipleMerges() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.SimpleMerge =
        props.class' <- "DefaultClass"
        props.draggable <- "stub"
        props.dir <- "ltr"
        div()

type [<Erase>] ComplexPropMerge() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.SimpleMerge =
        props.onKeyPress <- fun _ -> System.Console.WriteLine "Key pressed"
        div()

type [<Erase>] SimpleNestedMerge() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.whatever =
        div() {
            button() {
                props.class' <- "defaultClass"
            }
        }

type [<Erase>] NestedMerge() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.SimpleMerge =
        div() {
            props.dir <- "ltr"
            div()
            button() {
                props.class' <- "DefaultClass"
                props.draggable <- "stub"
            }
        }
        
type [<Erase>] NestedMergeWithChildren() =
    interface RegularNode
    [<SolidTypeComponent>]
    member props.SimpleMerge =
        div() {
            props.dir <- "ltr"
            div()
            button() {
                props.class' <- "DefaultClass"
                "Button"
                props.draggable <- "stub"
            }
        }
        

