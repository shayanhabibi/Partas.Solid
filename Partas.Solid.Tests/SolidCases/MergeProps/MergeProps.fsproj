<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net9.0</TargetFramework>
    </PropertyGroup>

    <ItemGroup>
        <Compile Include="MergeProps.fs" />
    </ItemGroup>

    <ItemGroup>
        <ProjectReference Include="..\..\..\Partas.Solid\Partas.Solid.fsproj" />
    </ItemGroup>

</Project>
