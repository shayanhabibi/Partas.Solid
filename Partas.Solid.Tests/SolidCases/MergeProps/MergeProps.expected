import { splitProps, mergeProps } from "solid-js";

export function SimpleMerge(props) {
    props = mergeProps({
        class: "DefaultClass",
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div />;
}

export function MultipleMerges(props) {
    props = mergeProps({
        class: "DefaultClass",
        draggable: "stub",
        dir: "ltr",
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div />;
}

export function ComplexPropMerge(props) {
    props = mergeProps({
        onKeyPress: (_arg) => {
            console.log("Key pressed");
        },
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div />;
}

export function SimpleNestedMerge(props) {
    props = mergeProps({
        class: "defaultClass",
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div>
        <button />
    </div>;
}

export function NestedMerge(props) {
    props = mergeProps({
        draggable: "stub",
        class: "DefaultClass",
        dir: "ltr",
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div>
        <div />
        <button />
    </div>;
}

export function NestedMergeWithChildren(props) {
    props = mergeProps({
        draggable: "stub",
        class: "DefaultClass",
        dir: "ltr",
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div>
        <div />
        <button>
            Button
        </button>
    </div>;
}

export function ValMutableMerge(props) {
    props = mergeProps({
        index: 5,
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div />;
}

export function ValMutableMergeInherited(props) {
    props = mergeProps({
        index: 5,
    }, props);
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    return <div />;
}

